php/site-utils.php

```diff
@@ -0,0 +1,274 @@
+<?php
+
+namespace EE\SiteUtils;
+
+use \EE;
+use \Symfony\Component\Filesystem\Filesystem;
+
+/**
+ * Get the site-name from the path from where ee is running if it is a valid site path.
+ *
+ * @return bool|String Name of the site or false in failure.
+ */
+function get_site_name() {
+	$sites = EE::db()::select( array( 'sitename' ) );
+
+	if ( $sites ) {
+		$cwd          = getcwd();
+		$name_in_path = explode( '/', $cwd );
+		$site_name    = array_intersect( EE\Utils\array_flatten( $sites ), $name_in_path );
+
+		if ( 1 === count( $site_name ) ) {
+			$name = reset( $site_name );
+			$path = EE::db()::select( array( 'site_path' ), array( 'sitename' => $name ) );
+			if ( $path ) {
+				$site_path = $path[0]['site_path'];
+				if ( $site_path === substr( $cwd, 0, strlen( $site_path ) ) ) {
+					return $name;
+				}
+			}
+		}
+	}
+
+	return false;
+}
+
+/**
+ * Function to set the site-name in the args when ee is running in a site folder and the site-name has not been passed in the args. If the site-name could not be found it will throw an error.
+ *
+ * @param array   $args     The passed arguments.
+ * @param String  $command  The command passing the arguments to auto-detect site-name.
+ * @param String  $function The function passing the arguments to auto-detect site-name.
+ * @param integer $arg_pos  Argument position where Site-name will be present.
+ *
+ * @return array Arguments with site-name set.
+ */
+function auto_site_name( $args, $command, $function, $arg_pos = 0 ) {
+	if ( isset( $args[$arg_pos] ) ) {
+		if ( EE::db()::site_in_db( $args[$arg_pos] ) ) {
+			return $args;
+		}
+	}
+	$site_name = get_site_name();
+	if ( $site_name ) {
+		if ( isset( $args[$arg_pos] ) ) {
+			EE::error( $args[$arg_pos] . " is not a valid site-name. Did you mean `ee $command $function $site_name`?" );
+		}
+		array_splice( $args, $arg_pos, 0, $site_name );
+	} else {
+		EE::error( "Could not find the site you wish to run $command $function command on.\nEither pass it as an argument: `ee $command $function <site-name>` \nor run `ee $command $function` from inside the site folder." );
+	}
+
+	return $args;
+}
+
+
+/**
+ * Function to check all the required configurations needed to create the site.
+ *
+ * Boots up the container if it is stopped or not running.
+ */
+function init_checks() {
+	$proxy_type = EE_PROXY_TYPE;
+	if ( 'running' !== EE::docker()::container_status( $proxy_type ) ) {
+		/**
+		 * Checking ports.
+		 */
+		@fsockopen( 'localhost', 80, $port_80_exit_status );
```

Comment:
@rahulsprajapati Removing suppression generates the following PHP warning:

```bash
PHP Warning:  fsockopen(): unable to connect to localhost:443 (Connection refused) in /root/workspace/easyengine/php/site-utils.php on line 79
PHP Stack trace:
PHP   1. {main}() /root/workspace/easyengine/php/boot-fs.php:0
PHP   2. include_once() /root/workspace/easyengine/php/boot-fs.php:17
PHP   3. EE\bootstrap() /root/workspace/easyengine/php/init-ee.php:24
PHP   4. EE\Bootstrap\LaunchRunner->process() /root/workspace/easyengine/php/bootstrap.php:77
PHP   5. EE\Runner->start() /root/workspace/easyengine/php/EE/Bootstrap/LaunchRunner.php:23
PHP   6. EE\Runner->_run_command_and_exit() /root/workspace/easyengine/php/EE/Runner.php:792
PHP   7. EE\Runner->run_command() /root/workspace/easyengine/php/EE/Runner.php:288
PHP   8. EE\Dispatcher\Subcommand->invoke() /root/workspace/easyengine/php/EE/Runner.php:265
PHP   9. call_user_func:{/root/workspace/easyengine/php/EE/Dispatcher/Subcommand.php:425}() /root/workspace/easyengine/php/EE/Dispatcher/Subcommand.php:425
PHP  10. EE\Dispatcher\CommandFactory::EE\Dispatcher\{closure}() /root/workspace/easyengine/php/EE/Dispatcher/Subcommand.php:425
PHP  11. call_user_func:{/root/workspace/easyengine/php/EE/Dispatcher/CommandFactory.php:96}() /root/workspace/easyengine/php/EE/Dispatcher/CommandFactory.php:96
PHP  12. Site_Command->create() /root/workspace/easyengine/php/EE/Dispatcher/CommandFactory.php:96
PHP  13. EE\SiteUtils\init_checks() /root/workspace/easyengine/vendor/easyengine/site-command/src/Site_Command.php:82
PHP  14. fsockopen() /root/workspace/easyengine/php/site-utils.php:79
```

Upon execution, it gives `bool(false)` but also prints this warning. Even try-catch block doesn't work here. The solution occurring to me is either suppression or temporarily changing error reporting here. Can there be any better handling or approach for this? Please advise what can be the best course for the same.

Reply:
ok. Just check once if we can archive this with php curl request or any alternative method to check port. Just wanted to avoid doing suppress here. 
If other methods are not that efficient and meet our requirement here then keep this as it is.

Reply:
@rahulsprajapati Checked this out with @mbtamuli. In php curl using:

> curl_setopt($ch, CURLOPT_PORT, $port);
$ip = curl_getinfo( $ch, CURLINFO_PRIMARY_IP );

Gives `$ip` as an empty string if the port is free and as `127.0.0.1` if the port is occupied. This can be used in our case.